scripts/build_locally.py

```diff
@@ -14,225 +14,206 @@
+        help="Use default oneAPI compiler layout",
     )
-    driver.add_argument(
+    p.add_argument(
         "--debug",
-        default="Release",
+        dest="build_type",
         const="Debug",
         action="store_const",
-        help="Set the compilation mode to debugging",
+        default="Release",
+        help="Set build type to Debug (defaults to Release)",
     )
-    driver.add_argument(
-        "--compiler-root",
-        type=str,
-        help="Path to compiler home directory",
-        default=None,
+
+    p.add_argument(
+        "--generator", type=str, default="Ninja", help="CMake generator"
     )
-    driver.add_argument(
+    p.add_argument(
         "--cmake-executable",
         type=str,
-        help="Path to cmake executable",
         default=None,
+        help="Path to CMake executable used by build",
     )
-    driver.add_argument(
-        "--no-level-zero",
-        help="Enable Level Zero support",
-        dest="level_zero",
-        action="store_false",
-    )
-    driver.add_argument(
+
+    p.add_argument(
         "--glog",
-        help="DPCTLSyclInterface uses Google logger",
         dest="glog",
         action="store_true",
+        help="Enable DPCTL Google logger support",
     )
-    driver.add_argument(
+    p.add_argument(
         "--verbose",
-        help="Build using vebose makefile mode",
         dest="verbose",
         action="store_true",
+        help="Enable verbose makefile output",
     )
-    driver.add_argument(
+
+    p.add_argument(
+        "--no-level-zero",
+        dest="no_level_zero",
+        action="store_true",
+        default=False,
+        help="Disable Level Zero backend (deprecated: use --target-level-zero "
+        "OFF)",
+    )
+    p.add_argument(
+        "--target-level-zero",
+        action="store_true",
+        help="Enable Level Zero backend explicitly",
+    )
+
+    p.add_argument(
         "--cmake-opts",
-        help="Options to pass through to cmake",
-        dest="cmake_opts",
-        default="",
         type=str,
+        default="",
+        help="Additional options to pass directly to CMake",
     )
-    driver.add_argument(
+
+    p.add_argument(
         "--target-cuda",
         nargs="?",
         const="ON",
-        help="Enable CUDA target for build; "
-        "optionally specify architecture (e.g., --target-cuda=sm_80)",
         default=None,
-        type=str,
+        help="Enable CUDA build. Architecture is optional to specify.",
     )
-    driver.add_argument(
+    p.add_argument(
         "--target-hip",
         required=False,
-        help="Enable HIP target for build. "
-        "Must specify HIP architecture (e.g., --target-hip=gfx90a)",
         type=str,
+        help="Enable HIP backend. Architecture required to be specified.",
     )
-    args = parser.parse_args()
 
-    args_to_validate = [
-        "c_compiler",
-        "cxx_compiler",
-        "compiler_root",
-    ]
+    p.add_argument(
```

Comment:
It does not work
```
$ python scripts/build_locally.py --clean
$ ls | grep _skbuild
> _skbuild

$ ls _skbuild/linux-x86_64-3.13/
> cmake-build  cmake-install  setuptools
```

Reply:
it seemed to work for me locally, though it didn't actually print the message that it was cleaning, strangely. If I don't activate oneAPI in the environment but pass `--clean`, the folder is gone. It's just remade quickly afterward with the compiler active.

Can you try this approach?

Reply:
tested again and it did seem to print and clean the directory